src/reducers/alerts.js

```diff
@@ -2,29 +2,47 @@ import alertsData from '../lib/alerts/index.jsx';
 import {AlertTypes, AlertLevels} from '../lib/alerts/index.jsx';
 import extensionData from '../lib/libraries/extensions/index.jsx';
 
-const SHOW_STANDARD_ALERT = 'scratch-gui/alerts/SHOW_STANDARD_ALERT';
+const SHOW_ALERT = 'scratch-gui/alerts/SHOW_ALERT';
 const SHOW_EXTENSION_ALERT = 'scratch-gui/alerts/SHOW_EXTENSION_ALERT';
 const CLOSE_ALERT = 'scratch-gui/alerts/CLOSE_ALERT';
 const CLOSE_ALERTS_WITH_ID = 'scratch-gui/alerts/CLOSE_ALERTS_WITH_ID';
 
+/**
+ * Initial state of alerts reducer
+ *
+ * {bool} visible - whether the alerts are visible
+ * {array} alertsList - list of alerts, each with properties:
+ *  * alert type (required): one of AlertTypes
+ *  * closeButton (optional): bool indicating that we should show close button
+ *  * content (optional): react element (a <FormattedMessage />)
+ *  * extentionId (optional): id string that identifies the extension
+ *  * iconURL (optional): string
+ *  * level (required): string, one of AlertLevels
+ *  * message (optional): string
+ *  * showReconnect (optional): bool
+ */
```

Comment:
This isn't valid jsdoc, but I don't think the linter cares because it doesn't know you're intending it to be :)  Previously I mentioned using jsdoc because there is already syntax for denoting required vs optional properties, properties of objects, etc. I think you can leave it (other than the line that doesn't show the actual key name, `alertType`) — we just need it documented, and this does the job well enough for people reading it. 

If you wanted it to be valid, I think you would need two different `@typedef`'s, one for `AlertsInitialState` and one for either `AlertsList` or just `Alert` (and `alertsList` would be `{Array.<Alert>}`). And then finally you would use `@type {AlertsInitialState}` above `initialState`.

Reply:
Yeah, I tried to research how you document this kind of stuff, and came up with nothing but the most vague examples.